py/objnamedtuple.c

```diff
@@ -109,7 +109,14 @@ STATIC mp_obj_t namedtuple_make_new(const mp_obj_type_t *type_in, size_t n_args,
     }
 
     // Create a tuple and set the type to this namedtuple
-    mp_obj_tuple_t *tuple = MP_OBJ_TO_PTR(mp_obj_new_tuple(num_fields, NULL));
+    mp_obj_tuple_t *tuple;
+    if (num_fields == 0) {
+        // The empty tuple is a constant object; explicit allocation needed
+        tuple = m_new_obj_var(mp_obj_tuple_t, mp_obj_t, 0);
```

Comment:
Does `m_new_obj_far` work for the non-empty case too? If yes, we can avoid the if statement.

Reply:
It should work for the non-empty case too, as it is what `mp_obj_new_tuple` calls internally.

This is my first contribution; what is the best way to act? Should I force-push that change to my branch, create a new branch or create a new commit?

Reply:
Since this is a small change, I would just force-push to this branch.